### PR TITLE
Support torch.bool in torch.sort + CUDA

### PR DESCRIPTION
**Pull request by xw285cornell**
Summary: This might be out-dated, so I'm adding it back and see if we pass all the tests. I'm pretty sure cuda12 is ok.

Test Plan: CI

Differential Revision: D65282650

### CODE REVIEW

**Review by zou3519 — APPROVED** [review]

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -18477,11 +18480,18 @@ def sample_inputs_alias_copy(op_info, device, dtype, requires_grad, **kwargs):
            )),
     OpInfo('sort',
            dtypes=all_types_and(torch.bool, torch.float16, torch.bfloat16),
-           dtypesIfCUDA=all_types_and(torch.float16, torch.bfloat16),
+           dtypesIfCUDA=all_types_and(torch.bool, torch.float16, torch.bfloat16),
            sample_inputs_func=sample_inputs_sort,
            supports_forward_ad=True,
            supports_fwgrad_bwgrad=True,
            skips=(
+                DecorateInfo(
```
Comment (xw285cornell) [inline torch/testing/_internal/common_methods_invocations.py:18488]: Need to add this because the non-standard bools (uint8) will matter for the sort order (at least for the indices). Wondering if there are some better ideas

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -3320,7 +3320,10 @@ def large_1d_unique():
     flag = [True, False]
     for dim, descending, stable in product(dims, flag, flag):
         # default schema without stable sort
-        yield SampleInput(small_3d_unique(), dim, descending)
+        if not (dtype == torch.bool and torch.device(device).type == 'cuda'):
+            # bool and cuda requires stable sort for stable results, at least
```
Comment (xw285cornell) [inline torch/testing/_internal/common_methods_invocations.py:3324]: adding this because we need stable sort (otherwise it'll fail inductor run which runs on the compiled graph and eager mode and produce different indices). We have the stable sort below and they run fine

**Review by xw285cornell — COMMENTED** [review]
@ngimel @eqy wondering if you can take a look and see if those changes are ok

**Review by ngimel — APPROVED** [review]
lgtm if the tests agree

File: torch/testing/_internal/common_methods_invocations.py (hunk 2 above)
Comment (nikitaved) [reply torch/testing/_internal/common_methods_invocations.py:3324]: The samples seem to be coming from `small_3d_unique`. Isn't it stable in this case?

Comment (xw285cornell) [reply torch/testing/_internal/common_methods_invocations.py:3324]: small_3d_unique only gives dim and descending but no stable flat. L3344-3346 explicitly set stable=True.

Comment (eqy) [reply torch/testing/_internal/common_methods_invocations.py:3324]: Would it be possible to account for this in the test cases rather than eliminating some combinations of inputs?

File: torch/testing/_internal/common_methods_invocations.py (hunk 1 above)
Comment (eqy) [reply torch/testing/_internal/common_methods_invocations.py:18488]: I'm OK with this if the behavior is determined by e.g., cub and we do not care to put in a workaround

**Review by eqy — APPROVED** [review]
